File: scala/src/main/java/Student.java
/**
 * @author name 婉然从物
 * @create 2023-10-21 15:43
 */
public class Student {
    private String name;
    private Integer age;
    private static String school = "atguigu";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo(){
        System.out.println(this.name + " " + this.age + " " + school);
    }
}

File: scala/src/main/Scala/chapter01/Student.scala
package chapter01

/**
 * @author name 婉然从物
 * @create 2023-10-21 15:51
 */
class Student(name: String, var age: Int) {
  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
  }

}

//  引入伴生对象
object Student{
  val school = "atguigu"

  def main(args: Array[String]): Unit = {
    val bob = new Student("bob", 20)
    val alice = new Student("alice", 23)

    alice.printInfo()
  }
}